python/ECR_REPOSITORY_SCAN_ON_PUSH_CHECK/ECR_REPOSITORY_SCAN_ON_PUSH_CHECK.py

```diff
@@ -0,0 +1,452 @@
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ecr_client = get_client('ecr', event)
+
+    repository_list_of_dict = get_all_repositories(ecr_client)
+    result = []
+    if not repository_list_of_dict:
+        return build_evaluation(
+            event['accountId'],
+            'NOT_APPLICABLE',
+            event,
+            resource_type='AWS::::Account'
+        )
+
+    for repository_dict in repository_list_of_dict:
+        if repository_dict['imageScanningConfiguration']['scanOnPush']:
+            result.append(build_evaluation(repository_dict['repositoryArn'], 'COMPLIANT', event))
+        else:
+            result.append(build_evaluation(
+                repository_dict['repositoryArn'],
+                'NON_COMPLIANT',
+                event,
+                annotation="The Amazon Elastic Container Registry repository is not configured to scan images on push."
+            ))
+    return result
+
+def get_all_repositories(ecr_client):
+    next_token = None
+    result_list = []
+
+    while True:
+        if next_token:
+            response_list_repositories_dict = ecr_client.describe_repositories(nextToken=next_token)
+        else:
+            response_list_repositories_dict = ecr_client.describe_repositories()
+
+        result_list.extend(response_list_repositories_dict['repositories'])
+        if 'nextToken' in response_list_repositories_dict:
+            next_token = response_list_repositories_dict['nextToken']
+        else:
+            return result_list
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    valid_rule_parameters = rule_parameters
```

Comment:
Are you doing any validation checks here?

Reply:
I wasn't sure what to check for. The rule isn't designed to take in a parameter.

I was just looking at another rule that does something similar (i.e. obtain a list of resources of a single type and iterate over each one to make sure that a given option is enabled) and found this one: https://github.com/awslabs/aws-config-rules/blob/master/python/EKS_LOGGING_CHECK/EKS_LOGGING_CHECK.py and it looks like they are doing something similar:
```python
def evaluate_parameters(rule_parameters):
    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
    Return:
    anything suitable for the evaluate_compliance()
    Keyword arguments:
    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
    """
    valid_rule_parameters = rule_parameters
    return valid_rule_parameters
```

Let me know if you have any suggestions. Should I be validating that no parameters have been passed in?

Reply:
it would be good to add another evaluation that raises a ValueError if there are any parameters. 

```
def parameters_exist(parameters):
    return len(parameters) != 0
```

Reply:
I added a check for parameters in the rule and included a test to assert an error when that happens.